include/cuda/std/detail/libcxx/include/__mdspan/trait_backports.hpp

```diff
@@ -0,0 +1,162 @@
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Christian R. Trott (crtrott@sandia.gov)
+//
+// ************************************************************************
+//@HEADER
+*/
+
+#ifndef _LIBCUDACXX___MDSPAN_TRAIT_BACKPORTS_HPP
+#define _LIBCUDACXX___MDSPAN_TRAIT_BACKPORTS_HPP
+
+#include "macros.hpp"
+#include "config.hpp"
+
+#ifndef __cuda_std__
+#include <type_traits>
+#include <utility> // integer_sequence
+#endif
+
+//==============================================================================
+// <editor-fold desc="Variable template trait backports (e.g., is_void_v)"> {{{1
+
+#ifdef __MDSPAN_NEEDS_TRAIT_VARIABLE_TEMPLATE_BACKPORTS
+
+#if __MDSPAN_USE_VARIABLE_TEMPLATES
+namespace std {
+
+#define __MDSPAN_BACKPORT_TRAIT(__TRAIT) \
+  template <class... _Args> __MDSPAN_INLINE_VARIABLE constexpr auto __TRAIT##_v = __TRAIT<_Args...>::value;
+
+__MDSPAN_BACKPORT_TRAIT(is_assignable)
+__MDSPAN_BACKPORT_TRAIT(is_constructible)
+__MDSPAN_BACKPORT_TRAIT(is_convertible)
+__MDSPAN_BACKPORT_TRAIT(is_default_constructible)
+__MDSPAN_BACKPORT_TRAIT(is_trivially_destructible)
+__MDSPAN_BACKPORT_TRAIT(is_same)
+__MDSPAN_BACKPORT_TRAIT(is_empty)
+__MDSPAN_BACKPORT_TRAIT(is_void)
+
+#undef __MDSPAN_BACKPORT_TRAIT
+
+} // end namespace std
+
+#endif // __MDSPAN_USE_VARIABLE_TEMPLATES
+
+#endif // __MDSPAN_NEEDS_TRAIT_VARIABLE_TEMPLATE_BACKPORTS
+
+// </editor-fold> end Variable template trait backports (e.g., is_void_v) }}}1
+//==============================================================================
+
+//==============================================================================
+// <editor-fold desc="integer sequence (ugh...)"> {{{1
+
+#if !defined(__MDSPAN_USE_INTEGER_SEQUENCE) || !__MDSPAN_USE_INTEGER_SEQUENCE
+
+namespace std {
+
+template <class _Tp, _Tp... _Vals>
+struct integer_sequence {
+  static constexpr std::size_t size() noexcept { return sizeof...(_Vals); }
+  using value_type = _Tp;
+};
+
+template <std::size_t... _Vals>
+using index_sequence = std::integer_sequence<std::size_t, _Vals...>;
+
+namespace __detail {
+
+template <class _Tp, _Tp _Np, _Tp _Ip, class _Result>
+struct __make_int_seq_impl;
+
+template <class _Tp, _Tp _Np, _Tp... _Vals>
+struct __make_int_seq_impl<_Tp, _Np, _Np, integer_sequence<_Tp, _Vals...>>
+{
+  using type = integer_sequence<_Tp, _Vals...>;
+};
+
+template <class _Tp, _Tp _Np, _Tp _Ip, _Tp... _Vals>
+struct __make_int_seq_impl<
+  _Tp, _Np, _Ip, integer_sequence<_Tp, _Vals...>
+> : __make_int_seq_impl<_Tp, _Np, _Ip+1, integer_sequence<_Tp, _Vals..., _Ip>>
+{ };
+
+} // end namespace __detail
+
+template <class _Tp, _Tp _Np>
+using make_integer_sequence = typename __detail::__make_int_seq_impl<_Tp, _Np, 0, integer_sequence<_Tp>>::type;
```

Comment:
I believe those are alreayd available at C++14 so Could you check whether we actually need this file at all?

Reply:
@miscco This is probably a left-over from when the reference mdspan implementation had to support C++11.  It should be possible to remove this.

Reply:
This file if removed since we don't need C++11 support.